o1vm/src/pickles/main.rs

```diff
@@ -48,10 +48,37 @@ pub fn cannon_main(args: cli::cannon::RunArgs) {
     let start = Start::create(state.step as usize);
 
     let domain_fp = EvaluationDomains::<Fp>::create(DOMAIN_SIZE).unwrap();
-    let srs: SRS<Vesta> = {
-        let srs = SRS::create(DOMAIN_SIZE);
-        srs.get_lagrange_basis(domain_fp.d1);
-        srs
+    let srs: SRS<Vesta> = match &args.srs_cache {
+        Some(cache) => {
+            debug!("Loading SRS from cache {}", cache);
```

Comment:
I would check that the SRS is of the expected size (`DOMAIN_SIZE`).

Reply:
Isn't the requirement that `srs.size() >= DOMAIN_SIZE` ?  because I'm looking that the size of `Vesta` srs and it's actually `2^16`, even though our domain size is half that

Reply:
The mismatch in the hard-coded `DOMAIN_SIZE` and what you can load from cache was responsible for a pretty big bug we were experiencing when trying to verify proofs for small programs. 

I made changes to remove the `const` and instead to use the info coming from the cache (if provided). I also changed the `1<<15` to `1<<16` to match what I was finding in the cache generated by the repo.